pylint/checkers/variables.py

```diff
@@ -1353,8 +1353,7 @@ def visit_name(self, node: nodes.Name | nodes.AssignName | nodes.DelName) -> Non
             return
 
         self._undefined_and_used_before_checker(node, stmt)
-        if self._is_undefined_loop_variable_enabled:
```

Comment:
Can't we also remove the class attribute? It's always going to be a problem for useless suppression, right ?

Reply:
Yeah! Good catch.

Reply:
Let's see if this also works. I haven't run tests locally but I think this should work.